File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int i,n=nums.size();
        int mid,l=0,r=n-1;
        while(l<=r)
        {
            mid=(l+r)/2;
            if(nums[mid]<target){
                l=mid+1;
            }
            else if(nums[mid]>target)
                r=mid-1;
            else
                return mid;
        }
        return -1;

    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashMap<Integer,Integer> cache = new HashMap<>();
        int i,n=nums.length;
        for(i=0;i<n;i++){
            if(cache.getOrDefault(nums[i], 0)>0)
                return true;
            cache.put(nums[i], cache.getOrDefault(nums[i], 0)+1);
        }
        return false;
    }
}